linera-service/src/linera.rs

```diff
@@ -1835,6 +1841,60 @@ impl Runnable for Job {
                 _ => unreachable!("other project commands do not require storage"),
             },
 
+            Wallet(WalletCommand::Init {
+                faucet: Some(faucet_url),
+                ..
+            }) => {
+                info!("Requesting a new chain from the faucet.");
+                let key_pair = context.generate_key_pair();
+                let public_key = key_pair.public();
+                context.wallet_state.add_unassigned_key_pair(key_pair);
+                let outcome = cli_wrappers::Faucet::claim_url(&public_key, &faucet_url).await?;
+                println!("{}", outcome.chain_id);
+                println!("{}", outcome.message_id);
+                println!("{}", outcome.certificate_hash);
+
+                let state = WorkerState::new("Local node".to_string(), None, storage)
+                    .with_allow_inactive_chains(true)
+                    .with_allow_messages_from_deprecated_epochs(true);
+                let mut node_client = LocalNodeClient::new(state, Notifier::default());
+
+                let admin_chain_id = context.wallet_state.genesis_admin_chain();
+                let query = ChainInfoQuery::new(admin_chain_id).with_committees();
+                let info = node_client.handle_chain_info_query(query).await?;
+                let committee = info
+                    .latest_committee()
+                    .context("Invalid chain info response; missing latest committee")?;
+                let nodes = context.make_node_provider().make_nodes(committee)?;
+
+                // Download the parent chain.
```

Comment:
Should we share the code with Assign?

Reply:
How come with have to download the parent chain? (and not the ancestor chains)

Reply:
Not downloading the parent chain will require a few more changes, e.g. to `local_node`. I'll do that in a separate PR.